biapol_utilities/label/_match_labels.py

```diff
@@ -1,88 +1,104 @@
-            
-            iou_streshold: float, optional
-                Threshold value above which two labels are accepted as overlapping.
-                The default value is 0.25
+    *metric*: callable, optional
+        Method to be used to generate a metric of similarity between labels
+        in subsequent slices. Must return a matrix with
+        `shape=(max(n, m), max(n, m))`, where n and m are the present labels
+        in two subsequent slices z and z+1. The values in the matrix must be
+        normalized to a range of [0, 1], where 0 and signify minimal or maximal
+        similarity between two labels.
+        Default is `intersection_over_union_matrix`
+    *filter*:ccallable, optional
+        Method to be used to filter values from the similarity matrix. This can
+        help to speed up the matching process if, fo instance, entries in the
+        similarity matrix below a defined threshold are set to zero. Default is
+        `suppressed_similarity(similiarity_matrix, threshold=0.3)`
+    *matching* : callable, optional
+        Method to be used for matching the labels. This function is supposed
+        to return a binary matrix with `shape=(n+1, m+1)` corresponding to
+        `match=1`, `no_match=0` of the `n` labels in a given slice and
+        `m` labels in the following slice. The default is `max_similarity`.
 
     Returns
     -------
     nd-array
-        Processed version of imageB with labels corresponding to imageA.
+        Processed version of label_image_y with labels corresponding to
+        label_image_x.
     """
-    
-    if method == intersection_over_union_matrix:
-        threshold = kwargs.get('iou_threshold', 0.25)
-    
+
+    method_metric = kwargs.get('metric', intersection_over_union_matrix)
+    method_filter = kwargs.get('filter', suppressed_similarity)
+    method_matching = kwargs.get('matching', max_similarity)
+
     # relabel label_image_y to keep overlap matrix small
     label_image_y, _, _ = relabel_sequential(label_image_y)
-    
-    
-    # Calculate image similarity matrix img_sim based on chosen method
-    img_sim = method(label_image_y, label_image_x)[1:,1:]
-    mmax = label_image_x.max()
-    
-    if img_sim.size > 0:
-        
-        # Keep only ious above threshold
-        img_sim[img_sim < threshold] = 0.0
-        img_sim[img_sim < img_sim.max(axis=0)] = 0.0
-        
-        # Pick value with highest IoU value
-        istitch = img_sim.argmax(axis=1) + 1
-        ino = np.nonzero(img_sim.max(axis=1) == 0.0)[0]  # Find unpaired labels
-        
-        # append unmatched labels and background to lookup table
-        istitch[ino] = np.arange(mmax+1, mmax+len(ino)+1, 1, int)  
-        mmax += len(ino)
-        istitch = np.append(np.array(0), istitch)
-        
-        return istitch[label_image_y]
+
+    # Calculate image similarity metric
+    similarity_matrix = method_metric(label_image_y.flatten(),
```

Comment:
Why do you flatten here? Is it the same as `ravel()`? Did you think consider the performance? See also:
https://stackoverflow.com/questions/28930465/what-is-the-difference-between-flatten-and-ravel-functions-in-numpy

Reply:
To be honest, I always used both interchangeably, but `ravel()` would probably be a better choice if we work with large arrays. Thanks for the link, I changed it in c8f3473410eae69d145b54b0bfbf174e4f7f5de6